doc/voq/aggregate_voq_counters.md

```diff
@@ -0,0 +1,99 @@
+
+
+# Aggregate VOQ Counters in SONiC #
+#### Rev 1.0
+
+## Table of Content 
+   * [Revision](#revision)
+   * [Overview](#overview)
+   * [Requirements](#requirements)
+   * [Architecture Design](#architecture-design)
+   * [High-Level Design](#high-level-design)
+      * [CHASSIS_APP_DB Changes](#chassis_app_db-changes)
+      * [SWSS Changes](#swss-changes)
+      * [Repositories that need to be changed](#repositories-that-need-to-be-changed)
+   * [SAI API](#sai-api)
+   * [Configuration and management](#configuration-and-management)
+      * [CLI](#cli)
+   * [Restrictions/Limitations](#restrictionslimitations)
+   * [Testing Requirements/Design](#testing-requirementsdesign)
+      * [System Test cases](#system-test-cases)  
+
+### Revision 
+| Rev |     Date    |       Author                                                                       | Change Description                |
+|:---:|:-----------:|:----------------------------------------------------------------------------------:|-----------------------------------|
+| 1.0 | 11-Jan-2024 | Harsis Yadav, Pandurangan R S, Vivek Kumar Verma (Arista Networks)               | Initial public version            | 
+
+### Overview 
+
+In a [distributed VOQ architecture](https://github.com/sonic-net/SONiC/blob/master/doc/voq/architecture.md) corresponding to each output VOQ present on an ASIC, there are VOQs present on every ASIC in the system. Each ASIC has its own set of VOQ stats maintained in the FSI which needs to be gathered independently and can be hard to visualize, providing a non-cohesive experience.
+
+### Requirements
+
+Provide aggregate VOQ counters in a distributed VOQ architecture.
+
+### Architecture Design 
+
+No new architecture changes are required to SONiC. An instance of SWSS runs for each ASIC on it's respective FSI instance, in a distributed VOQ architecture. SWSS can be used to poll VOQ stats for every ASIC and update CHASSIS_APP_DB which is accessible from every FSI module. 
+
+### High-Level Design
+
+Figure 1: Gathering the VOQ stats in CHASSIS_APP_DB
+![Sequence Diagram](images/add_voq_seq.png "Figure 1: Sequence Diagram")  
+Figure 2: Aggregation of VOQ stats
+![Aggregation of VOQ Stats](images/add_voq_cli.png "Figure 2: Aggregation of VOQ Stats")
+
+
+#### CHASSIS_APP_DB Changes
+
+The following new VOQ counters should be available for each VOQ entry in the DB:
+   * `COUNTERS|fsi_id|asic_id|intf@fsi_id|asic_id:VOQ_index`
```

Comment:
Can you add an example here for an entry?

Reply:
@vivekverma-arista please share key as well as images are not getting loaded here.

Reply:
@vmittal-msft you should be able to see the images when viewing the files. For example, see https://github.com/sonic-net/SONiC/blob/f4d03c5efb0c40fc3159c52d49918b1724b111b1/doc/voq/aggregate_voq_counters.md